feat(nut): Add new socket timeout support, introduces Co…

AdamLeyshon opened the pull request:
…nnection Timeout and Operation Timeout parameters to the client connect method, defaults to 10 and 30 seconds respectively.
Resolves #8

AdamLeyshon committed:
feat(nut): Add new socket timeout support, resolves #8

AdamLeyshon committed:
chore: Update connect() documentation

AdamLeyshon committed:
chore: Correct `go.mod`

AdamLeyshon commented:
Sorry to bug you all again, any chance we can get this merged? It's blocking a PR with adding functionality to Telegraf
Thanks in advance!

robbiet480 commented on example_test.go:
```diff
@@ -2,17 +2,16 @@ package nut
 import (
 	"fmt"
-
-	nut "github.com/robbiet480/go.nut"
```
Can you revert these changes? The example_test.go file needs them as is.

AdamLeyshon replied:
Done

AdamLeyshon committed:
Update example_test.go

As requested in https://github.com/robbiet480/go.nut/pull/9#pullrequestreview-961396719
